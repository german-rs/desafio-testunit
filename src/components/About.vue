<template>
    <main class="container">
        <section>
            <h2>Â¿QuÃ© es Vue?</h2>
            <p>Vue.js es un framework progresivo de JavaScript para 
                construir interfaces de usuario. Su enfoque en la 
                simplicidad y flexibilidad lo hace ideal tanto para 
                pequeÃ±os proyectos como para aplicaciones complejas.
            </p>
            <img src="https://via.placeholder.com/400x200" alt="Framework Vue.js">
        </section>

        <section>
            <h2>Â¿QuÃ© es Vitest?</h2>
            <p>Vitest es un framework moderno para pruebas, optimizado para proyectos que utilizan Vite. Ofrece un entorno rÃ¡pido, sencillo de configurar y con excelentes capacidades para pruebas unitarias y de integraciÃ³n.</p>
            <img src="https://via.placeholder.com/400x200" alt="Herramienta Vitest">
        </section>

        <section>
            <h2>Â¿Por quÃ© usar Vue y Vitest juntos?</h2>
            <div class="div-box">
                <div>
                    <h3>Velocidad <span>âš¡</span> </h3>
                    <p>Vitest aprovecha las caracterÃ­sticas de Vite, ofreciendo una experiencia de prueba ultrarrÃ¡pida.</p>
                </div>
                <div>
                    <h3>Compatibilidad <span>ðŸ”§</span> </h3>
                    <p>Vitest incluye herramientas especÃ­ficas para trabajar con componentes Vue sin complicaciones.</p>
                </div>
                <div>
                    <h3>Ecosistema <span>ðŸ“¦</span> </h3>
                    <p>Ambos forman parte de un ecosistema de desarrollo moderno, optimizado para desarrolladores actuales.</p>
                </div>
            </div>
        </section>
    </main>
</template>

<style scoped>
p{
    text-align: center;
    max-width: 600px;
}
section{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: .5em;
    margin-bottom: 3em;
}
h3{
    padding: 0.5em;
    text-align: center;
}

.div-box{
    display: flex;
    flex-direction: column;
    padding: 1em;
    gap: 2em;
}
</style>