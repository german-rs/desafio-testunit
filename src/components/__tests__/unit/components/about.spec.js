import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import About from '@/views/AboutView.vue';

describe('About.vue', () => {
  it('Debe coincidir con el snapshot', () => {
    const wrapper = mount(About);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Debe tener texto alternativo para todas las imágenes', () => {
    const wrapper = mount(About);
    const images = wrapper.findAll('img');
    
    images.forEach(img => {
      expect(img.attributes('alt')).toBeTruthy();
      expect(img.attributes('alt').trim()).not.toHaveLength(0);
    });
  });

  it('Los títulos de las secciones deben ser etiquetas h2', () => {
    const wrapper = mount(About);
    const sections = wrapper.findAll('section');
    
    sections.forEach(section => {
      const h2 = section.find('h2');
      expect(h2.exists()).toBe(true);
    });
  });
});
