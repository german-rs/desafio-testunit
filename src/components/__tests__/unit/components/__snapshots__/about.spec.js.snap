// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About.vue > Debe coincidir con el snapshot 1`] = `
"<div data-v-d4948df9="">
  <header data-v-d4948df9="" class="container">
    <h1 data-v-d4948df9="">Acerca de Vue y Vitest</h1>
    <p data-v-d4948df9="">Descubre cÃ³mo Vue y Vitest trabajan juntos para facilitar el desarrollo y las pruebas de tus aplicaciones web. </p>
  </header>
  <main data-v-161b4f7e="" data-v-d4948df9="" class="container">
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿QuÃ© es Vue?</h2>
      <p data-v-161b4f7e="">Vue.js es un framework progresivo de JavaScript para construir interfaces de usuario. Su enfoque en la simplicidad y flexibilidad lo hace ideal tanto para pequeÃ±os proyectos como para aplicaciones complejas. </p><img src="https://via.placeholder.com/400x200" alt="Framework Vue.js" data-v-161b4f7e="">
    </section>
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿QuÃ© es Vitest?</h2>
      <p data-v-161b4f7e="">Vitest es un framework moderno para pruebas, optimizado para proyectos que utilizan Vite. Ofrece un entorno rÃ¡pido, sencillo de configurar y con excelentes capacidades para pruebas unitarias y de integraciÃ³n.</p><img src="https://via.placeholder.com/400x200" alt="Herramienta Vitest" data-v-161b4f7e="">
    </section>
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿Por quÃ© usar Vue y Vitest juntos?</h2>
      <div class="div-box" data-v-161b4f7e="">
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Velocidad <span data-v-161b4f7e="">âš¡</span></h3>
          <p data-v-161b4f7e="">Vitest aprovecha las caracterÃ­sticas de Vite, ofreciendo una experiencia de prueba ultrarrÃ¡pida.</p>
        </div>
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Compatibilidad <span data-v-161b4f7e="">ğŸ”§</span></h3>
          <p data-v-161b4f7e="">Vitest incluye herramientas especÃ­ficas para trabajar con componentes Vue sin complicaciones.</p>
        </div>
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Ecosistema <span data-v-161b4f7e="">ğŸ“¦</span></h3>
          <p data-v-161b4f7e="">Ambos forman parte de un ecosistema de desarrollo moderno, optimizado para desarrolladores actuales.</p>
        </div>
      </div>
    </section>
  </main>
</div>"
`;

exports[`About.vue > debe coincidir con el snapshot 1`] = `
"<div data-v-d4948df9="">
  <header data-v-d4948df9="" class="container">
    <h1 data-v-d4948df9="">Acerca de Vue y Vitest</h1>
    <p data-v-d4948df9="">Descubre cÃ³mo Vue y Vitest trabajan juntos para facilitar el desarrollo y las pruebas de tus aplicaciones web. </p>
  </header>
  <main data-v-161b4f7e="" data-v-d4948df9="" class="container">
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿QuÃ© es Vue?</h2>
      <p data-v-161b4f7e="">Vue.js es un framework progresivo de JavaScript para construir interfaces de usuario. Su enfoque en la simplicidad y flexibilidad lo hace ideal tanto para pequeÃ±os proyectos como para aplicaciones complejas. </p><img src="https://via.placeholder.com/400x200" alt="Framework Vue.js" data-v-161b4f7e="">
    </section>
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿QuÃ© es Vitest?</h2>
      <p data-v-161b4f7e="">Vitest es un framework moderno para pruebas, optimizado para proyectos que utilizan Vite. Ofrece un entorno rÃ¡pido, sencillo de configurar y con excelentes capacidades para pruebas unitarias y de integraciÃ³n.</p><img src="https://via.placeholder.com/400x200" alt="Herramienta Vitest" data-v-161b4f7e="">
    </section>
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿Por quÃ© usar Vue y Vitest juntos?</h2>
      <div class="div-box" data-v-161b4f7e="">
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Velocidad <span data-v-161b4f7e="">âš¡</span></h3>
          <p data-v-161b4f7e="">Vitest aprovecha las caracterÃ­sticas de Vite, ofreciendo una experiencia de prueba ultrarrÃ¡pida.</p>
        </div>
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Compatibilidad <span data-v-161b4f7e="">ğŸ”§</span></h3>
          <p data-v-161b4f7e="">Vitest incluye herramientas especÃ­ficas para trabajar con componentes Vue sin complicaciones.</p>
        </div>
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Ecosistema <span data-v-161b4f7e="">ğŸ“¦</span></h3>
          <p data-v-161b4f7e="">Ambos forman parte de un ecosistema de desarrollo moderno, optimizado para desarrolladores actuales.</p>
        </div>
      </div>
    </section>
  </main>
</div>"
`;

exports[`About.vue > should match the snapshot 1`] = `
"<div data-v-d4948df9="">
  <header data-v-d4948df9="" class="container">
    <h1 data-v-d4948df9="">Acerca de Vue y Vitest</h1>
    <p data-v-d4948df9="">Descubre cÃ³mo Vue y Vitest trabajan juntos para facilitar el desarrollo y las pruebas de tus aplicaciones web. </p>
  </header>
  <main data-v-161b4f7e="" data-v-d4948df9="" class="container">
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿QuÃ© es Vue?</h2>
      <p data-v-161b4f7e="">Vue.js es un framework progresivo de JavaScript para construir interfaces de usuario. Su enfoque en la simplicidad y flexibilidad lo hace ideal tanto para pequeÃ±os proyectos como para aplicaciones complejas. </p><img src="https://via.placeholder.com/400x200" alt="Framework Vue.js" data-v-161b4f7e="">
    </section>
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿QuÃ© es Vitest?</h2>
      <p data-v-161b4f7e="">Vitest es un framework moderno para pruebas, optimizado para proyectos que utilizan Vite. Ofrece un entorno rÃ¡pido, sencillo de configurar y con excelentes capacidades para pruebas unitarias y de integraciÃ³n.</p><img src="https://via.placeholder.com/400x200" alt="Herramienta Vitest" data-v-161b4f7e="">
    </section>
    <section data-v-161b4f7e="">
      <h2 data-v-161b4f7e="">Â¿Por quÃ© usar Vue y Vitest juntos?</h2>
      <div class="div-box" data-v-161b4f7e="">
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Velocidad <span data-v-161b4f7e="">âš¡</span></h3>
          <p data-v-161b4f7e="">Vitest aprovecha las caracterÃ­sticas de Vite, ofreciendo una experiencia de prueba ultrarrÃ¡pida.</p>
        </div>
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Compatibilidad <span data-v-161b4f7e="">ğŸ”§</span></h3>
          <p data-v-161b4f7e="">Vitest incluye herramientas especÃ­ficas para trabajar con componentes Vue sin complicaciones.</p>
        </div>
        <div data-v-161b4f7e="">
          <h3 data-v-161b4f7e="">Ecosistema <span data-v-161b4f7e="">ğŸ“¦</span></h3>
          <p data-v-161b4f7e="">Ambos forman parte de un ecosistema de desarrollo moderno, optimizado para desarrolladores actuales.</p>
        </div>
      </div>
    </section>
  </main>
</div>"
`;
